tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs

```diff
@@ -839,8 +863,70 @@ private async void WidgetControl_Drop(object sender, DragEventArgs e)
         }
 
         ViewModel.PinnedWidgets.CollectionChanged += OnPinnedWidgetsCollectionChangedAsync;
+    }
 
-        _log.Debug($"Drop ended");
+    /// <summary>
+    /// Handle keyboard shortcuts for moving widgets left and right.
+    /// </summary>
+    private async void HandleKeyUpAsync(object sender, KeyRoutedEventArgs e)
+    {
+        _log.Debug($"Key up");
+
+        await _moveWidgetsLock.WaitAsync();
+        try
+        {
+            var key = e.Key;
+            _log.Debug($"e.Key = {key}");
+            if (e.Handled || !(key == VirtualKey.Left || key == VirtualKey.Right))
+            {
+                return;
+            }
+
+            var focusedItem = e.OriginalSource as GridViewItem;
+            if (focusedItem == null || focusedItem.Content == null || focusedItem.Content is not WidgetViewModel widgetViewModel)
+            {
+                return;
+            }
+
+            if (IsAltAndShiftPressed())
+            {
+                var index = ViewModel.PinnedWidgets.IndexOf(widgetViewModel);
+                _log.Information($"Move widget {widgetViewModel.WidgetDisplayTitle} at index {index} {key}");
+
+                if (key == VirtualKey.Left && index > 0)
+                {
+                    await MoveWidgetAsync(widgetViewModel, index, index - 1);
+                    await FocusManager.TryFocusAsync(WidgetGridView.ItemsPanelRoot.Children.ElementAt(index - 1), FocusState.Keyboard);
+                    _log.Debug($"Focus moved to index {index - 1}");
+                }
+                else if (key == VirtualKey.Right && index < (ViewModel.PinnedWidgets.Count - 1))
+                {
+                    // Setting focus before and after the move looks more natural than letting the focus move to the wrong element.
+                    await FocusManager.TryFocusAsync(WidgetGridView.ItemsPanelRoot.Children.ElementAt(index + 1), FocusState.Keyboard);

```

Comment:
Why the extra focus here, but not in the if case?

Reply:
There was nothing I could set focus on in the if case that would make the animation look less awkward. Someday I'd like better animations for both keyboard and the drag and drop scenario, but I'd rather unblock accessibility asap than hold anything up for that.